### ISSUE TITLE
Issue with CustomExternalForce going from 7.4.1 to 7.7.0

### ISSUE THREAD

**Issue by LeoFHall** [openmm/openmm#4497]
Hi !
I'm having an issue with running a piece of code on the computacanada clusters, that use openmm 7.7.0, but it runs fine with 7.4.2 which I have on my laptop

it's with this function:

```
#Adding a force to restrain atoms
restraint = CustomExternalForce('k*periodicdistance(x,y,z,x0,y0,z0)^2')
restraint.addGlobalParameter('k',kspring*kilojoules_per_mole/nanometer)
restraint.addPerParticleParameter('x0') #for each particle a reference position is determined
restraint.addPerParticleParameter('y0')
restraint.addPerParticleParameter('z0')
for atom in topology.atoms(): #Apply restraint to all the particles except the fixed one
    if atom.index != FixedParticle:
        print(pdb.positions[atom.index])
        restraint.addParticle(atom.index,pdb.positions[atom.index]) 
system.addForce(restraint)
```
when I test it on the cluster I get this error :
```
Traceback (most recent call last):
  File "toy-MeOH-DeltaA1-6666p67-150K.py", line 87, in <module>
    restraint.addParticle(atom.index,pdb.positions[atom.index]) #pdb.positions[i] gives the desired reference position ?
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc9/cuda11.4/openmpi4/openmm/7.7.0/lib/python3.8/site-packages/openmm/openmm.py", line 9650, in addParticle
    return _openmm.CustomExternalForce_addParticle(self, *args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'CustomExternalForce_addParticle'.
  Possible C/C++ prototypes are:
    OpenMM::CustomExternalForce::addParticle(int,std::vector< double,std::allocator< double > > const &)
    OpenMM::CustomExternalForce::addParticle(int)
```

the print(pdb.position[atom.index] returns
`Vec3(x=1.5235, y=1.7182, z=1.8938) nm
`
I don't know C but this looks like what's expected in the function, and I don't really understand what is if not.
I tried running the code with 
`        restraint.addParticle(atom.index) 
`
in the loop but that just complains there's the wrong number of parametres.
Thanks for the help
Leo

**Comment by peastman** [comment]
Can you test the most recent release (8.1.1)?  The following works for me.

```python
from openmm import *
from openmm.unit import *
f = CustomExternalForce('0')
f.addPerParticleParameter('x0')
f.addPerParticleParameter('y0')
f.addPerParticleParameter('z0')
pos = Vec3(0, 0, 0)*nanometers
f.addParticle(0, pos)
```

**Comment by LeoFHall** [comment]
8.1.1 isn't available on computecanada clusters but 8.0.0 is and that worked so problem solved I guess ^^

Thanks !
